src/components/TodoList/TodoList.tsx

```diff
@@ -1,208 +1,100 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoFeat } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
+import { Status } from '../../types/Status';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useAppDispatch();
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const filter = useAppSelector(state => state.filter);
+  const todos = useAppSelector(state => state.todos.filter(
+    (todo: Todo) => {
+      return todo.title.toLocaleLowerCase().includes(filter.query.toLocaleLowerCase());
+    },
+  ));
+
+  const isSelected = (id: number) => currentTodo?.id === id;
+
+  const visibleTodos = useMemo(() => {
+    switch (filter.status) {
+      case Status.All:
+        return todos.filter((todo: Todo) => todo);
```

Comment:
todos list doesn't include null or undefined, so I don't any sense in filtering todos on Status.All
for this case, default case is enough